### PR TITLE
Fix stepped list in dark mode didn't show numbers

### LINKED ISSUE #4656: Bug: Stepped List in dark mode didn't show numbers
**Describe the bug**

In the dark theme stepped list numbers in circles are overwritten with white background.

Screenshot 2023-02-14 at 12 44 16

**To Reproduce**

Switch system setting to dark theme:

Create an empty project with the vanilla framework and add a stepped list.

**Expected behavior**

Number in white color, a circle with white border and transparent background.

Screenshot 2023-02-14 at 12 48 34

**Possible solution**

For `.p-stepped-list__title::before` in dark mode change to `background-color` to `transparent` and `border-color` to `rgba(255, 255, 255, 0.87)`.

**Desktop (please complete the following information):**

- OS: MacOS 13.1
- Browser firefox, chrome
- Version latest

**Smartphone (please complete the following information):**

Haven't tested

### PR DESCRIPTION
**Pull request by abhigyanghosh30**
## Done

Fixed issue of numbers not being displayed correctly on stepped lists

Fixes #4656 

## QA

- Open [demo](https://vanilla-framework-4674.demos.haus/docs/examples/patterns/lists/lists-stepped-dark)
- The numbers in the title should be white over a dark background
- Review updated documentation:
  - No update in new classnames 

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

## Screenshots

image

### CODE REVIEW

**Comment by abhigyanghosh30** [comment]
For this I was wondering I we could just make the background transparent for the circle. That way it would work for both as well as any other themes we might have in the future

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "vanilla-framework",
-  "version": "3.12.0",
+  "version": "3.12.1",
```
Comment (bartaz) [inline package.json:3]: We haven't released 3.12.0, so you don't need to do it. Sorry, I know it's not exactly clear when to do it.

File: scss/_patterns_lists.scss [hunk 2]
```diff
@@ -333,6 +333,10 @@ $list-status-icon-nudge: 0.3rem; // 0.3rem seems to be a magic number that posit
         width: $bullet-width;
       }
     }
+    .p-strip--dark &::before {
+      background-color: $color-x-dark;
```
Comment (bartaz) [inline scss/_patterns_lists.scss:337]: I guess let's use the same as dark strip

```suggestion
      background-color: $color-dark;
```

**Comment by bartaz** [comment]
As mentioned on standup, the standalone example needs fixing:

https://vanilla-framework-4674.demos.haus/docs/examples/standalone/patterns/lists/lists-stepped-dark

**Review by bartaz — APPROVED** [review]

**Pull request merged**